packages/extension/src/view/devtools/components/cookies/cookieTopBar/index.tsx

```diff
@@ -51,10 +64,82 @@ const CookieSearch = ({
     })
   );
 
-  const { getCookiesSetByJavascript } = useCookieStore(({ actions }) => ({
+  const cookieDeletedRef = useRef(false);
+  const selectedCookieIndexRef = useRef(-1);
+
+  const isAnyCookieSelected =
+    cookieDeletedRef.current && !selectedFrameCookie
+      ? filteredCookies.length > 0
+        ? true
+        : false
+      : selectedFrameCookie &&
+        selectedFrameCookie[Object.keys(selectedFrameCookie)[0]]
+      ? true
+      : false;
+
+  const {
+    deleteCookie,
+    deleteAllCookies,
+    getCookiesSetByJavascript,
+    selectedFrame,
+  } = useCookieStore(({ state, actions }) => ({
+    deleteCookie: actions.deleteCookie,
+    deleteAllCookies: actions.deleteAllCookies,
     getCookiesSetByJavascript: actions.getCookiesSetByJavascript,
+    selectedFrame: state.selectedFrame,
   }));
 
+  useEffect(() => {
+    cookieDeletedRef.current = false;
+  }, [selectedFrame]);
+
+  const handleDeleteCookie = useCallback(() => {
+    const selectedKey =
+      cookieDeletedRef.current && !selectedFrameCookie
+        ? filteredCookies[
+            getNextIndexToDelete(
+              selectedCookieIndexRef.current,
+              filteredCookies.length
+            )
+          ]
+        : Object.values(selectedFrameCookie ?? {})[0];
+
+    if (selectedKey !== null && selectedKey.parsedCookie) {
+      const cookieKey = getCookieKey(selectedKey?.parsedCookie);
+      if (cookieKey) {
+        selectedCookieIndexRef.current = filteredCookies.findIndex(
+          (cookie) => getCookieKey(cookie.parsedCookie) === cookieKey
+        );
+        deleteCookie(cookieKey);
+        cookieDeletedRef.current = true;
+        const index = getNextIndexToDelete(
+          selectedCookieIndexRef.current,
+          filteredCookies.length - 1
+        );
+        const tempRows = filteredCookies.filter((cookie) => {
```

Comment:
```suggestion
        const filteredRows = filteredCookies.filter((cookie) => {
```

Reply:
fixed in [9de5a3d](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/278/commits/9de5a3df8410e59c636e0d8210f8ff5dea009c17)